lightkurve/targetpixelfile.py

```diff
@@ -190,6 +195,88 @@ def from_archive(target, cadence='long', quarter=None, month=None,
     def __repr__(self):
         return('KeplerTargetPixelFile Object (ID: {})'.format(self.keplerid))
 
+
+    def get_sources(self, catalog=None, magnitude_limit=18, dist_tolerance=3):
+        """
+        Returns a crossmatched table of stars that are centered on the target
+        of the tpf file. Current catalog supported are KIC, EPIC and Gaia DR2.
+
+        Parameters
+        -----------
+        catalog: string
+            Indicate catalog assigned for mission. If Kepler, catalog will be KIC
+            if K2 catalog is EPIC.
+        magnitude_limit : int or float
+            Limit the returned magnitudes to only stars brighter than magnitude_limit.
+            Default 18.
+        dist_tolerance: float
+            Tolerance for source matching in pixels. Sources that fall within this
+            tolerance of pixels in the KeplerTargetPixelFile will be returned in the result.
+            Default is 3 pixels (12 arcsec).
+
+        Returns
+        -------
+        result : astropy.table
+            Astropy table with the following columns
+        ID : astropy.table.column (KIC & EPIC)
+            Catalog ID from catalog.
+        RAJ200: astropy.table.column (KIC & EPIC)
+            Right ascension [degrees]
+        DEJ2000: astropy.table.column (KIC & EPIC)
+            Declination [deg]
+        pmRA: astropy.table.column (KIC & EPIC)
+            Proper motion for right ascension [mas/year]
+        pmDEC: astropy.table.column (KIC & EPIC)
+        Kpmag: astropy.table.column (KIC & EPIC)
+            Magnitude in Kepler band [mag]
+        """
+
+        if catalog is None:
+            if self.mission == 'Kepler':
+                catalog = 'KIC'
+            elif self.mission == 'K2':
+                catalog = 'EPIC'
+            else:
+                log.error('Please provide a catalog.')
+
+        #Skycoord the centre of target
+        cent = SkyCoord(ra=self.ra, dec=self.dec, frame='icrs', unit=(u.deg, u.deg))
+
+        #Find the size of the TPF
+        radius = np.max(self.flux.shape[1:]) * 4 * u.arcsecond
+
+        # query around centre with radius
+        data = query_catalog(cent, radius=radius, catalog=catalog)
+
+        # Load ra & dec of all tpf pixels
+        pixels_ra, pixels_dec = self.get_coordinates(cadence='all')
+
+        # Reduce calculation for astroy seperation
+        pixel_radec = np.asarray([pixels_ra.ravel(), pixels_dec.ravel()])
+        pixel_radec = np.round(pixel_radec, decimals=5)
+
+        # Return unique pairs
+        pixel_pairs = (np.unique(pixel_radec, axis=1))
```

Comment:
This step can be dropped if we use `self.get_coordinates(cadence=0)` above.

Reply:
We are checking all the cadences for targets, including the motion. The `np.unique` above reduces the number of separations we need to calculate.

Reply:
Ohhh I see, you are using all pixel coordinates of all cadences by design.  What is the `%timeit` i.e. speed cost?

Reply:
We should only ever have to run this once for a given target, and I am guessing the cost of running this separation test is << than querying the catalog online. 

@AndyTza do you want to run a time on this? It's a good skill to practice. :)

Reply:
590 ms ± 13.2 ms per loop (mean ± std. dev. of 7 runs, 1 loop each) (__cadence='all'__)
208 ms ± 6.03 ms per loop (mean ± std. dev. of 7 runs, 1 loop each) (__cadence=0__)
1.5 s ± 35.7 ms per loop (mean ± std. dev. of 7 runs, 1 loop each) (__removed np.round()__)

Reply:
@barentsen @christinahedges For the time being,  I've kept cadence='all'. Moving forward we can discuss if we want to include this step.

Reply:
I'm not sure how we feel about this. Either way, I don't think it should be the first cadence, as that one is often a little off source.